Reagents added to microwave carry over to final product

emmanuelbassil opened the pull request:
🆑 emmanuelbassil, Azlan
refactor: Microwave recipes now transfer over non-consumed reagents to the final product; minimizing creation of reagents out of thin air and food items tasting different than the items that make it up.
rscadd: Added recipes for frosted donuts; no longer depending on random chance for them to be cooked.
rscadd: Made chaos donut significantly more chaotic; check the wiki for more details. Sprite by Azlan. 
rscadd: Added soy versions of dough, flat dough, dough slice, spaghetti, batter, cake batter, and bread. They can be substituted in all relevant recipes.
balance: Modified almost all recipes and their nutritional value to make them more consistent with each other.
balance: Alcohol now has some nutritious value instead of none. It also replenishes some hydration; but not as well as water.
balance: Alcohol now has sugar content; to the great chagrin of Unathi. 
balance: Juices now have very little nutritious value instead of none; from 10-40% of baseline nutriment depending on the juice.
tweak: Now able to frost any donut at any time by simply adding some sprinkles.
tweak: Added new microwave meat burger recipe (Meatball + Bun).
tweak: Spaghetti recipes can now be made with finished boiled spaghetti as an ingredients instead of water/raw spaghetti
fix: Burgers made by adding a bun to meat now have the same nutritional value as burgers made via microwave.
fix: Baked goods made with normal dough or its derivatives now correctly contain animal protein to account for batter’s egg content.
fix: Soy batter recipe now properly produces vegan batter instead of regular batter. The recipe now requires soy milk instead of regular milk.
/🆑 

In the goal of working to expand allergies to food; first step is to make reagents behave more consistently in cooking.
A small step is to have reagents used in microwave recipes actually carry over into the end item. A nice side effect is that food no longer magically salts itself when you don't actually add salt.

For instance; current behavior has two ways to make fried egg: 
1) One Egg + One Pepper + One Salt
2) Two Eggs

As fried egg is seasoned and proteinized on initialize; this means both recipes have the same exact nutritional value.
With this change; recipe one now comes seasoned and with 3 egg protein. On the other hand; recipe two has double the protein but is bland. Of course; it can be seasoned after being cooked.

Added another category of 'consumed_reagent' for a recipe that is not carried over into the final product; think water for boiled egg or polyacid for roasted dionae. 

I still need to go through every recipe for this PR to be considered complete. Effectively, this will let you create multiple recipes that will give the same end product but can still have any taste/nutritional profile you like.

emmanuelbassil committed:
Reagents added to microwave carry over to final product

Sbotkin commented:
Might result in a rare situation where reagents mix in food/people, unintentionally.

emmanuelbassil commented:
> Might result in a rare situation where reagents mix in food/people, unintentionally.

Will keep an eye out during testing. I don’t anticipate a lot of player facing change but I could be wrong.

emmanuelbassil committed:
Jelly donuts

Fixes

Styling

emmanuelbassil committed:
Meat & Chaos

More recipes

More items

Pizzas

Auto stash before rebase of "AllergyUpda"

More Items

REbase

Rebase2

Rebase

Rebase

SierraKomodo reviewed (dismissed):
Rebase

emmanuelbassil commented:
> Rebase

Rebased; also updated changelog with new changes. Hoping to finish this by the end of the week.

emmanuelbassil committed:
Vegans Unite

Expanded soy options

emmanuelbassil committed:
More Vegans

emmanuelbassil committed:
Cakes

emmanuelbassil commented:
Almost nearing the end of this. Unless some other wrench pops up; just about 50 recipes left to go through; but shouldn't generate much code to review other than balancing nutriment/reagent content to match everything else.

Word document will be provided separately detailing all content changes for wiki editors.

emmanuelbassil committed:
Rest of Recipes_microwave.dm

Cheeses

emmanuelbassil commented:
Ready for review. Will send word document of recipe changes to wiki team

emmanuelbassil committed:
More Nutriment Types

emmanuelbassil commented:
Implemented feedback from rootoo
Added new protein nutriment subtypes for cheese, fish, and shellfish. Did not implement allergies for them as of yet; but should  be trivial. (So no user facing changes)
Also modified some recipes as per feedback.
